### PR TITLE
Update governance to add contributor ladder and steering committee governance

### PR DESCRIPTION
**Pull request by gnossen**
This PR amends our project governance to bring us in line with the majority of CNCF projects. In particular, it:

- Adds a Steering Committee with clearly defined election procedures, open to individuals regardless of their employer
- Introduces a contributor ladder clearly defining how an individual / employee can go from zero presence in the project to a maintainer with full voting rights.

The added text in this PR is largely pulled from CNCF governance templates.

This PR bootstraps into this new process by:

- Moving existing project contributors to an appropriate rung within the new contributor ladder based on discussions with senior gRPC project members.
- Outlining the details of the election of the first Steering Committee.

### CODE REVIEW

File: governance.md [hunk 1]
```diff
@@ -77,4 +67,130 @@ Details on gRPC release process and schedule can be found [here](https://github.
...
+#### Eligibility to Vote
+
+gRPC project [Maintainers](contributor_ladder.md#maintainer), as defined by
+[the contributor ladder](contributor_ladder.md), are eligible to vote. The
+official list of Maintainers will be kept up to date in this repository in
+[contributors/maintainers.md](contributors/maintainers.md).
+
+Anyone holding status as a [Maintainer](contributor_ladder.md#maintainer) at or
+before the voting for an election has closed will be able to cast a vote. The
+Election Officer(s_ will make available a voter's guide within this repository.
```
Comment (ejona86) [inline governance.md:174]: (s), not underscore

File: governance.md [hunk 2]
```diff
@@ -77,4 +67,130 @@ Details on gRPC release process and schedule can be found [here](https://github.
...
+
+#### Eligibility to Vote
+
+gRPC project [Maintainers](contributor_ladder.md#maintainer), as defined by
+[the contributor ladder](contributor_ladder.md), are eligible to vote. The
+official list of Maintainers will be kept up to date in this repository in
+[contributors/maintainers.md](contributors/maintainers.md).
+
+Anyone holding status as a [Maintainer](contributor_ladder.md#maintainer) at or
+before the voting for an election has closed will be able to cast a vote. The
```
Comment (ejona86) [inline governance.md:173]: "before the voting for an election has closed" could be read to include no-longer Maintainers from a decade before. "Anyone holding status as a Maintainer at any point while the voting is open will be able to cast a vote"?

File: governance.md [hunk 3]
```diff
@@ -77,4 +67,130 @@ Details on gRPC release process and schedule can be found [here](https://github.
...
+Election Officer(s_ will make available a voter's guide within this repository.
+
+#### Voting Procedure
+
+Elections will be held using a time-limited
+[Condorcet](https://en.wikipedia.org/wiki/Condorcet_method) ranking using the
+IRV method. Voters will be able to rank the candidates from most to least
+preferred, possibly omitting candidates. The top vote-getters will be elected to
+the open seats.
```
Comment (ejona86) [inline governance.md:182]: How long is a steering member's term? If 1 year, then all seats will be "open" seats at the time of a regular election?

**Review by ejona86 — COMMENTED** [review]
Page 1

File: contributor_ladder.md [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+_multiple_ areas of expertise. This is especially true for
+[gRFCs](governance.md#the-grfc-process). In this case, _one_ maintainer acts as
+the triager and is responsible for adding a maintainer to the review to
+represent all relevant areas of expertise. For example, a new load balancing
+policy proposed in a gRFC must be implementable in all supported languages, so a
+Maintainer for C++, Go, Java, etc. should be added to the review by the triaging
+Maintainer.
+
+Process of becoming a maintainer:
+1. Any current Maintainer may nominate a current Reviewer to become a new Maintainer, by opening a PR to [maintainers.md](contributors/maintainers.md).
```
Comment (ejona86) [inline contributor_ladder.md]: s/Reviewer/Lieutenant/

**Review by ejona86 — COMMENTED** [review]
Page 2

File: governance.md [hunk 5]
```diff
@@ -77,4 +67,130 @@ Details on gRPC release process and schedule can be found [here](https://github.
...
+open to anyone.
+
+### Election Procedure
+
+#### Timeline
+
+Steering Committee elections are held annually and terms are a year in length.
+Six weeks or more before the election, the Steering Committee will appoint
+Election Officer(s) (see below).  Two weeks or more before the election, the
+Election Officer(s) will issue a call for nominations. Three days before the
```
Comment (ejona86) [inline governance.md:145]: In the voting doc, you talk about the requirements for a nominee to be added (anyone can add a nominee, but with consent). There's no seconding or such or show of support needed. Should that be in this doc to some degree?

Comment (gnossen) [reply governance.md:145]: Good point. I wanted to leave some of the details of voting out (i.e. how ballots are formed, how nominations are submitted), but something as abstract as consent definitely seems like something we'd want to add to the governance itself. Done.

**Pull request closed without merging**